pkg/controller/windowsmachineconfig/windowsmachineconfig_controller.go

```diff
@@ -192,13 +207,14 @@ func (r *ReconcileWindowsMachineConfig) Reconcile(request reconcile.Request) (re
 
 	// Get the current count of required number of Windows VMs
 	currentCountOfWindowsVMs := len(windowsNodes.Items)
-
-	// Update status to reflect that operator is reconciling
-	r.statusMgr.setStatusConditions([]wmcapi.WindowsMachineConfigCondition{
-		*wmcapi.NewWindowsMachineConfigCondition(wmcapi.Reconciling, corev1.ConditionTrue, "", "")})
-	if err = r.statusMgr.updateStatus(); err != nil {
-		// Don't continue reconciling and requeue on status error
-		return reconcile.Result{}, err
+	if currentCountOfWindowsVMs != int(instance.Spec.Replicas) {
+		// Update status to reflect that operator is reconciling
+		r.statusMgr.setStatusConditions([]wmcapi.WindowsMachineConfigCondition{
+			*wmcapi.NewWindowsMachineConfigCondition(wmcapi.Reconciling, corev1.ConditionTrue, "", "")})
+		if err = r.statusMgr.updateStatusReconciling(); err != nil {
+			// Don't continue reconciling and requeue on status error
+			return reconcile.Result{}, err
+		}
```

Comment:
Should this be done inside `reconcileWindowsNodes`?

Reply:
could be moved there, kept it here since all status sets are being done in reconcile